ambry-messageformat/src/main/java/com.github.ambry.messageformat/MessageFormatWriteSet.java

```diff
@@ -55,8 +50,12 @@ public MessageFormatWriteSet(InputStream streamToWrite, List<MessageInfo> stream
   public long writeTo(Write writeChannel)
       throws IOException {
     ReadableByteChannel readableByteChannel = Channels.newChannel(streamToWrite);
-    writeChannel.appendFrom(readableByteChannel, sizeToWrite);
-    return sizeToWrite;
+    long sizeWritten = 0;
```

Comment:
@pnarayanan Yes that remains the same because index entries are not written until all the whole write has succeeded.